File: RQ2_Prompt_Elements/GPT3.5_Data/HumanEvalJava_output/scenario3/id_26T2000Test.java
// RemoveDuplicates.java
package scenario3;

import java.util.ArrayList;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

class RemoveDuplicates {

    /**
     * From a list of integers, remove all elements that occur more than once.
     * Keep order of elements left the same as in the input.
     *
     * > removeDuplicates([1, 2, 3, 2, 4])
     * [1, 3, 4]
     * > removeDuplicates([])
     * []
     * > removeDuplicates([1, 2, 3, 4])
     * [1, 2, 3, 4]
     * > removeDuplicates([1, 2, 3, 2, 4, 3, 5])
     * [1, 4, 5]
     */
    public static List<Object> removeDuplicates(List<Object> numbers) {
        Set<Object> set = new LinkedHashSet<>(numbers);
        List<Object> result = new ArrayList<>(set);
        return result;
    }
}

// RemoveDuplicatesTest.java
package scenario3;

import java.util.Arrays;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link RemoveDuplicates}.
* It contains ten unit test cases for the {@link RemoveDuplicates#removeDuplicates(List)} method.
*/
class RemoveDuplicatesTest {

    @Test
    void testRemoveDuplicates() {
        List<Object> input1 = Arrays.asList(1, 2, 3, 2, 4);
        List<Object> expectedOutput1 = Arrays.asList(1, 3, 4);
        assertEquals(expectedOutput1, RemoveDuplicates.removeDuplicates(input1));

        List<Object> input2 = Arrays.asList();
        List<Object> expectedOutput2 = Arrays.asList();
        assertEquals(expectedOutput2, RemoveDuplicates.removeDuplicates(input2));

        List<Object> input3 = Arrays.asList(1, 2, 3, 4);
        List<Object> expectedOutput3 = Arrays.asList(1, 2, 3, 4);
        assertEquals(expectedOutput3, RemoveDuplicates.removeDuplicates(input3));

        List<Object> input4 = Arrays.asList(1, 2, 3, 2, 4, 3, 5);
        List<Object> expectedOutput4 = Arrays.asList(1, 4, 5);
        assertEquals(expectedOutput4, RemoveDuplicates.removeDuplicates(input4));

        List<Object> input5 = Arrays.asList(1, 1, 1, 1, 1);
        List<Object> expectedOutput5 = Arrays.asList(1);
        assertEquals(expectedOutput5, RemoveDuplicates.removeDuplicates(input5));

        List<Object> input6 = Arrays.asList(1, 2, 3, "a", "b", "c", "a", "b");
        List<Object> expectedOutput6 = Arrays.asList(1, 2, 3, "a", "b", "c");
        assertEquals(expectedOutput6, RemoveDuplicates.removeDuplicates(input6));

        List<Object> input7 = Arrays.asList("a", "b", "c", "a", "b", "c");
        List<Object> expectedOutput7 = Arrays.asList("a", "b", "c");
        assertEquals(expectedOutput7, RemoveDuplicates.removeDuplicates(input7));

        List<Object> input8 = Arrays.asList(1.0, 2.0, 3.0, 2.0, 4.0);
        List<Object> expectedOutput8 = Arrays.asList(1.0, 3.0, 4.0);
        assertEquals(expectedOutput8, RemoveDuplicates.removeDuplicates(input8));

        List<Object> input9 = Arrays.asList(1, 2, 3, 2, 4, 3, 5, null, null);
        List<Object> expectedOutput9 = Arrays.asList(1, 4, 5, null);
        assertEquals(expectedOutput9, RemoveDuplicates.removeDuplicates(input9));

        List<Object> input10 = Arrays.asList(null, null, null, null);
        List<Object> expectedOutput10 = Arrays.asList(null);
        assertEquals(expectedOutput10, RemoveDuplicates.removeDuplicates(input10));
    }
}

File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario3/id_26T2000Test.py
# id_26_Test.java
from scenario3.id_26 import remove_duplicates

import pytest

class id_26_Test:
    """
    Test class of id_26.py.
    It contains ten unit test cases for the :func:`scenario3.id_26.remove_duplicates`.
    """


    def test_remove_duplicates_1(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4]) == [1, 3, 4]

    def test_remove_duplicates_2(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10]) == [1, 3, 6, 7, 8, 9]

    def test_remove_duplicates_3(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10]) == [1, 3, 6, 7, 8, 9]

    def test_remove_duplicates_4(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10]) == [1, 3, 6, 7, 8, 9]

    def test_remove_duplicates_5(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11]) == [1, 3, 6, 7, 8, 9, 11]

    def test_remove_duplicates_6(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 11]) == [1, 3, 6, 7, 8, 9, 11]

    def test_remove_duplicates_7(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 11, 11]) == [1, 3, 6, 7, 8, 9, 11]

    def test_remove_duplicates_8(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 11, 11, 11]) == [1, 3, 6, 7, 8, 9, 11]

    def test_remove_duplicates_9(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 11, 11, 11, 12]) == [1, 3, 6, 7, 8, 9, 11, 12]

    def test_remove_duplicates_10(self):
        """
        Test case for the :func:`scenario3.id_26.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 11, 11, 11, 12, 12]) == [1, 3, 6, 7, 8, 9, 11, 12]

# id_27.py
from typing import List


def remove_duplicates(numbers: List[int]) -> List[int]:
    """ From a list of integers, remove all elements that occur more than once.
    Keep order of elements left the same as in the input.
    >>> remove_duplicates([1, 2, 3, 2, 4])
    [1, 3, 4]
    """

    pass

# id_27_Test.java
from scenario3.id_27 import remove_duplicates

import pytest

class id_27_Test:
    """
    Test class of id_27.py.
    It contains ten unit test cases for the :func:`scenario3.id_27.remove_duplicates`.
    """
		
    def test_remove_duplicates_1(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4]) == [1, 3, 4]

    def test_remove_duplicates_2(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10]) == [1, 3, 6, 7, 8, 9]

    def test_remove_duplicates_3(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10]) == [1, 3, 6, 7, 8, 9]

    def test_remove_duplicates_4(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10]) == [1, 3, 6, 7, 8, 9]

    def test_remove_duplicates_5(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11]) == [1, 3, 6, 7, 8, 9, 11]

    def test_remove_duplicates_6(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 11]) == [1, 3, 6, 7, 8, 9, 11]

    def test_remove_duplicates_7(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 11, 11]) == [1, 3, 6, 7, 8, 9, 11]

    def test_remove_duplicates_8(self):
        """
        Test case for the :func:`scenario3.id_27.remove_duplicates`.
        """
        assert remove_duplicates([1, 2, 3, 2, 4, 4, 5, 5, 5, 6, 7,